### PR TITLE
[Data Objects] [Inheritance] Values at child not getting inherited from parent.

### LINKED ISSUE #6338: [Data Objects] [Inheritance] Values at child not getting inherited from parent.
**Expected behavior**

A child object should be able to inherit the values from its parent.

**Actual behavior**

Fields are not shown marked as inherited in the UI but showing as copied instead.
Noticed inheritance issues on Number, QuantityValue and Select types.

Please see screenshot attached below.

**Steps to reproduce**

- Create a new class and allow inheritance.
- Add a text, number, quantityValue and select field in definition. 
- Create an object of the same class. Add some data to all the fields. Save or Save&Publish.
- Create a child object.
- The data for respective fields will not inherit but gets copied from parent to the child.

inheritance issue

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #6338

### CODE REVIEW

File: models/DataObject/Traits/DefaultValueTrait.php [hunk 1]
```diff
@@ -130,7 +117,27 @@ protected function handleDefaultValue($data, $object = null, $params = [])
...
+            $configuredDefaultValue = $this->doGetDefaultValue($object, $context);
+            if (!$this->isEmpty($configuredDefaultValue)) {
+
+                if ($class && $class->getAllowInherit()) {
+                    $params = [];
+
+                    try {
+                        $data = $owner->getValueFromParent($this->getName(), $params);
+                        if (!$this->isEmpty($data)) {
+                            return $data;
```
Comment (dvesh3) [inline models/DataObject/Traits/DefaultValueTrait.php:133]: if parent has value then it should return null so as to inherit the value form parent. right?

Comment (weisswurstkanone) [reply models/DataObject/Traits/DefaultValueTrait.php:133]: No. It should return the value from the parent as this will go directly into the resource

**Comment by dvesh3** [comment]
when using default value then null values does not show inherited value in UI even though db has correct values.
image

image

File: models/DataObject/Traits/DefaultValueTrait.php (hunk 1 above)
Comment (dvesh3) [reply models/DataObject/Traits/DefaultValueTrait.php:133]: please add a note in docs, if not there :)

**Pull request merged**